base/path.jl

```diff
@@ -63,17 +63,17 @@ Return the current user's home directory.
 """
 function homedir()
     path_max = 1024
-    buf = Vector{UInt8}(undef, path_max)
+    buf = Base.StringVector(path_max + 1)
```

Comment:
```suggestion
    buf = Base.StringVector(path_max)
```

StringVector is guaranteed to be null-terminated

Reply:
so `path_max` should be `pathmax = 1024 - 1` to hold the null?